packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -0,0 +1,237 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library amplify_api;
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api/src/native_api_plugin.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:async/async.dart';
+import 'package:flutter/services.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import 'amplify_api_config.dart';
+import 'amplify_authorization_rest_client.dart';
+
+/// {@template amplify_api.amplify_api_dart}
+/// The AWS implementation of the Amplify API category.
+/// {@endtemplate}
+class AmplifyAPIDart extends AmplifyAPI {
+  late final AWSApiPluginConfig _apiConfig;
+  final http.Client? _baseHttpClient;
+
+  /// A map of the keys from the Amplify API config to HTTP clients to use for
+  /// requests to that endpoint.
+  final Map<String, AmplifyAuthorizationRestClient> _clientPool = {};
+
+  /// The registered [APIAuthProvider] instances.
+  final Map<APIAuthorizationType, APIAuthProvider> _authProviders = {};
+
+  /// {@macro amplify_api.amplify_api_dart}
+  AmplifyAPIDart({
+    List<APIAuthProvider> authProviders = const [],
+    http.Client? baseHttpClient,
+    this.modelProvider,
+  })  : _baseHttpClient = baseHttpClient,
+        super.protected() {
+    authProviders.forEach(registerAuthProvider);
+  }
+
+  @override
+  Future<void> configure({AmplifyConfig? config}) async {
+    final apiConfig = config?.api?.awsPlugin;
+    if (apiConfig == null) {
+      throw const ApiException('No AWS API config found',
+          recoverySuggestion: 'Add API from the Amplify CLI. See '
+              'https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/flutter/#configure-api');
+    }
+    _apiConfig = apiConfig;
+  }
+
+  @override
+  Future<void> addPlugin() async {
+    final nativeBridge = NativeApiBridge();
+    try {
+      List<String> authProvidersList =
```

Comment:
```suggestion
      final authProvidersList =
```

Stylistic note: we use the following lints in other packages and it's what I'd like to move towards in all the categories, we just haven't implemented it here yet. But, it's easier to build good habits now, so that's why I call it out:

- [omit_local_type](https://dart.dev/tools/linter-rules#omit_local_variable_types)
- [prefer_final_locals](https://dart.dev/tools/linter-rules#prefer_final_locals)